packages/components/psammead-social-embed/src/index.jsx

```diff
@@ -23,23 +23,25 @@ export const CanonicalSocialEmbed = ({
   const isSupportedProvider = Object.keys(providers).includes(provider);
   const hasCaption = caption && caption.text;
 
-  if (isSupportedProvider && oEmbed && hasCaption)
-    return (
-      <SkipLinkWrapper service={service} provider={provider} {...skipLink}>
+  const renderCanonicalContent = () => {
+    if (isSupportedProvider && oEmbed && hasCaption)
+      return (
         <CaptionWrapper service={service} {...caption}>
           <CanonicalEmbed provider={provider} oEmbed={oEmbed} />
         </CaptionWrapper>
-      </SkipLinkWrapper>
-    );
+      );
 
-  if (isSupportedProvider && oEmbed)
-    return (
-      <SkipLinkWrapper service={service} provider={provider} {...skipLink}>
-        <CanonicalEmbed provider={provider} oEmbed={oEmbed} />
-      </SkipLinkWrapper>
-    );
+    if (isSupportedProvider && oEmbed)
+      return <CanonicalEmbed provider={provider} oEmbed={oEmbed} />;
 
-  return <Notice service={service} provider={provider} {...fallback} />;
+    return <Notice service={service} provider={provider} {...fallback} />;
+  };
+
+  return (
+    <SkipLinkWrapper service={service} provider={provider} {...skipLink}>
+      {renderCanonicalContent()}
```

Comment:
Why do we need to declare a function and call it here rather than just rendering in the `<SkipLinkWrapper>` tag?

Reply:
This was keeping semi-complex control logic out of JSX to improve readability, but it obviously wasn't helping, so I've refactored it. Hopefully this makes more sense.